src/Psalm/Type/Union.php

```diff
@@ -41,7 +40,6 @@
  */
 final class Union implements TypeNode, Stringable
 {
-    use ImmutableNonCloneableTrait;
```

Comment:
Why this change?
Full Union immutability was recently introduced by one of my PRs, since the class is marked as `psalm-immutable` it makes no sense to clone it (and after I removed all clones in a later-merged PR performance improved by some 15%)

Reply:
That's great! Sadly that PR failed our end-to-end tests, because PSL apparently has a Psalm plugin that calls `Union::__clone`. [Not sure how to work around that](https://github.com/php-standard-library/psalm-plugin/blob/af8580e603772c9aad47ca6272d33ae15fc0502a/src/EventHandler/Type/Optional/FunctionReturnTypeProvider.php#L31) — @azjezz do you have ideas?

Reply:
In the meantime I'm going to merge this — we can always add the trait back once the end-to-end test has been fixed.

Reply:
@azjezz Fixing this is actually pretty simple, just replacing the last three lines with `return $argument_type->setPossiblyUndefined(true)` :)

Reply:
@danog Thanks! will do so :)

Reply:
@azjezz You should generally also remove all `clone`s of Unions and atomics from your plugin code :)

Reply:
(Please note that this is only valid for Psalm v5, Psalm v4 still requires clones and your legacy logic)

Reply:
ops! didn't think of that :D not released broken 2.0.3 :D

Reply:
I actually already did this in https://github.com/vimeo/psalm/pull/8697 (basically provided a custom patch file for our version), this was just a heads-up of the required changes for v5 :)